share/completions/doas.fish

```diff
@@ -14,7 +14,7 @@ function __fish_doas_print_remaining_args
     # we want.
     if test -n "$argv"
         and not string match -qr '^-' $argv[1]
-        echo $argv
+        string escape -- $argv
```

Comment:
Some kind of unescaping is needed, try typing `sudo clang -#` and hit Tab.
WIth this, clang completions think the commandline is `clang -\#` instead of `clang -#`, that's why it doesn't work.
Sadly `string unescape` will split on (admittedly rare) newlines, so `eval` is probably the way to go.

Reply:
Yep, `clang.fish` complete way not such as `gcc.fish`.
`sudo gcc -#` work, but clang not work...
We need `eval` or `join0 split0`

Thanks for your help~